src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,110 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.List;
+import duke.task.Task;
+
+/**
+ * Encapsulates the concept of a list of tasks.
+ */
+public class TaskList {
+    private List<Task> tasks;
+
+    /**
+     * Creates a TaskList with no tasks stored.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
```

Comment:
```suggestion
        tasks = new ArrayList<>();
```

Should the `this` specifier be removed as [it is not shadowed by a method or constructor parameter](https://se-education.org/guides/conventions/java/index.html#variables)?